openshift_resources/oadp/data_protection_application.py

```diff
@@ -1,17 +1,516 @@
+    )
+    searches: Optional[List[str]] = Field(
+        None,
+        description='A list of DNS search domains for host-name lookup. This will be appended to the base search paths generated from DNSPolicy. Duplicated search paths will be removed.',
+    )
+
+
+class SnapshotLocationVelero(BaseModel):
+    config: Optional[Dict[str, str]] = Field(
+        None, description='Config is for provider-specific configuration fields.'
+    )
+    provider: str = Field(
+        ..., description='Provider is the provider of the volume storage.'
+    )
+
+
+class SnapshotLocation(BaseModel):
+    velero: SnapshotLocationVelero = Field(
+        ...,
+        description='VolumeSnapshotLocationSpec defines the specification for a Velero VolumeSnapshotLocation.',
+    )
+
+
+class DataProtectionApplicationSpecModel(BaseModel):
+    backupImages: Optional[bool] = Field(
+        None,
+        description='backupImages is used to specify whether you want to deploy a registry for enabling backup and restore of images',
+    )
+    backupLocations: Optional[List[BackupLocation]] = Field(
+        None,
+        description='backupLocations defines the list of desired configuration to use for BackupStorageLocations',
+    )
+    configuration: DataProtectionApplicationConfiguration = Field(
+        ...,
+        description="configuration is used to configure the data protection application's server config",
+    )
+    features: Optional[Features] = Field(
+        None,
+        description='features defines the configuration for the DPA to enable the OADP tech preview features',
+    )
+    podAnnotations: Optional[Dict[str, str]] = Field(
+        None, description='add annotations to pods deployed by operator'
+    )
+    podDnsConfig: Optional[PodDnsConfig] = Field(
+        None,
+        description='podDnsConfig defines the DNS parameters of a pod in addition to those generated from DNSPolicy. https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-dns-config',
+    )
+    podDnsPolicy: Optional[str] = Field(
+        None,
+        description="podDnsPolicy defines how a pod's DNS will be configured. https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy",
+    )
+    snapshotLocations: Optional[List[SnapshotLocation]] = Field(
+        None,
+        description='snapshotLocations defines the list of desired configuration to use for VolumeSnapshotLocations',
+    )
+    unsupportedOverrides: Optional[Dict[str, str]] = Field(
+        None,
+        description='unsupportedOverrides can be used to override images used in deployments. Available keys are:   - veleroImageFqin   - awsPluginImageFqin   - openshiftPluginImageFqin   - azurePluginImageFqin   - gcpPluginImageFqin   - csiPluginImageFqin   - dataMoverImageFqin   - resticRestoreImageFqin   - kubevirtPluginImageFqin',
+    )
+
+
+class DataProtectionApplicationConditionConditionStatus(Enum):
```

Comment:
ConditionStatus is suppose to be defined in [core v1](https://github.com/oadp-qe/openshift-resources/blob/main/src/resources/io/k8s/api/core/v1.py), if it's missing there, let's add it there and remove from here

https://pkg.go.dev/k8s.io/api/core/v1#ConditionStatus